src/mono/wasm/runtime/driver.c

```diff
@@ -491,6 +491,9 @@ mono_wasm_string_from_js (const char *str)
 static int
 class_is_task (MonoClass *klass)
 {
+	if (!strcmp ("AsyncStateMachineBox`1", mono_class_get_name (klass)))
```

Comment:
I'd rather avoid hardcoding all classes which derive from Task even if it's unlikely there will be many.

@vargaz @lambdageek is there a cheap way to check for MonoClass to derive from Task?

Reply:
> I'd rather avoid hardcoding all classes which derive from Task even if it's unlikely there will be many.

Yeah, this is very fragile. There are others that can be returned as well.

Reply:
Recent commit should be better uses `is_subclass_of`.

Local tests show that it works for `Task`, ```Task`1``` and ```AsyncStateMachineBox`1```.